### ISSUE TITLE
File attachment error when using PHP7.1

### ISSUE THREAD

**Issue by Graham-72** [backdrop-contrib/views_send#14]
This error appears when changing from PHP7.0 to 7.1.

Error: [] operator not supported for strings in views_send_form_submit() (line 645 of ....../httpdocs/backdrop/modules/views_send/views_send.module).

**Comment by hansfn** [comment]
Yes, this is a backward incompatible change in PHP 7.1 - see "The empty index operator is not supported for strings anymore" on http://php.net/manual/en/migration71.incompatible.php I'll have to apply the fix for the Drupal 7 version too.

When you introduce this fix, you could also remove line 791 

`$attachments = isset($params['views_send_attachments']) ? $params['views_send_attachments'] : array();`

since you have made sure that the attachment always is an array.

**Comment by Graham-72** [comment]
@hansfn thanks for this.

**Comment by hansfn** [comment]
@Graham-72 And thanks to you. Regarding my comment about line 791: It is only true if you move your fix outside the if-block - before line 630.

**Comment by Graham-72** [comment]
Re-opened in order to apply code change as suggested, which is in step with D7 version.

**Comment by Graham-72** [comment]
>When you introduce this fix, you could also remove line 791

I am finding that I need something here to assign a value to `$attachments`. Another small commit will fix this.
